### PR TITLE
Use nameAllocator for enums in both Java/Kotlin

### PR DESCRIPTION
**Pull request by oldergod**
I messed up the rebase it seems but only 3 last commits are actually in the diff.
- Generate tests : b630e43
- Update tests : 0e46439
- Use nameAllocator for enums in both Java/Kotlin : c92aa0c

### CODE REVIEW

File: wire-library/wire-gson-support/src/test/java/com/squareup/wire/GsonTest.java [hunk 1]
```diff
@@ -66,26 +68,58 @@
...
+                KeywordJava.KeywordJavaEnum.final_,
+                KeywordJava.KeywordJavaEnum.public_,
+                KeywordJava.KeywordJavaEnum.package_,
+                KeywordJava.KeywordJavaEnum.return_
+            )
+        )
+        .build();
     String json = gson.toJson(keyword);
-    assertJsonEquals(json, "{\"final\":\"final\", \"public\":true, \"package\":{}, \"return\":[]}");
+    // TODO(benoit) We should print the constants as defined in the proto schema.
```
Comment (swankjesse) [inline wire-library/wire-gson-support/src/test/java/com/squareup/wire/GsonTest.java:113]: put the correct expectations and `@Ignore` the test

File: wire-library/wire-gson-support/src/test/java/com/squareup/wire/GsonTest.java [hunk 2]
```diff
@@ -66,26 +68,58 @@
...
+                KeywordKotlin.KeywordKotlinEnum.fun_,
+                KeywordKotlin.KeywordKotlinEnum.return_
+                // Broken, see https://github.com/square/kotlinpoet/issues/991
+                // KeywordKotlin.KeywordKotlinEnum.open
+            )
+        )
+        .build();
     String json = gson.toJson(keyword);
-    assertJsonEquals(json, "{\"object\":\"object\",\"when\":1, \"fun\":{}, \"return\":[]}");
+    // TODO(benoit) We should print the constants as defined in the proto schema.
```
Comment (swankjesse) [inline wire-library/wire-gson-support/src/test/java/com/squareup/wire/GsonTest.java:86]: put the correct expectations and @Ignore the test

File: wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 3]
```diff
@@ -562,11 +562,12 @@ public TypeSpec generateEnum(EnumType type) {
         constantBuilder.addAnnotation(Deprecated.class);
       }
 
-      builder.addEnumConstant(constant.getName(), constantBuilder.build());
+      builder.addEnumConstant(nameAllocator.get(constant), constantBuilder.build());
```
Comment (swankjesse) [inline wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java:565]: preallocate these with `newName` ?

File: wire-library/wire-moshi-adapter/src/test/java/com/squareup/wire/MoshiTest.java [hunk 4]
```diff
@@ -115,27 +116,58 @@
...
+                KeywordKotlin.KeywordKotlinEnum.return_
+                // Broken, see https://github.com/square/kotlinpoet/issues/991
+                // KeywordKotlin.KeywordKotlinEnum.open
+            )
+        )
+        .build();
     String json = adapter.toJson(keyword);
-    JsonUtils.assertJsonEquals(json,
-        "{\"object\":\"object\",\"when\":1, \"fun\":{}, \"return\":[]}");
+    // TODO(benoit) We should print the constants as defined in the proto schema.
```
Comment (swankjesse) [inline wire-library/wire-moshi-adapter/src/test/java/com/squareup/wire/MoshiTest.java:134]: same

File: wire-library/wire-moshi-adapter/src/test/java/com/squareup/wire/MoshiTest.java [hunk 5]
```diff
@@ -115,27 +116,58 @@
...
+                KeywordJava.KeywordJavaEnum.public_,
+                KeywordJava.KeywordJavaEnum.package_,
+                KeywordJava.KeywordJavaEnum.return_
+            )
+        )
+        .build();
     String json = adapter.toJson(keyword);
-    JsonUtils.assertJsonEquals(json,
-        "{\"final\":\"final\", \"public\":true, \"package\":{}, \"return\":[]}");
+    // TODO(benoit) We should print the constants as defined in the proto schema.
```
Comment (swankjesse) [inline wire-library/wire-moshi-adapter/src/test/java/com/squareup/wire/MoshiTest.java:162]: same

**Review by swankjesse — APPROVED** [review]

File: wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java (hunk 3 above)
Comment (oldergod) [reply wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java:565]: They're already inthere that's why I can just look them up passing the constant.
See https://github.com/square/wire/blob/master/wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java#L226-L228

**Pull request merged**